### PR TITLE
Possible refactor of _update_attrs_and_maybe_adapt_lights

### PR DESCRIPTION
**Pull request by basnijholt**
@th3w1zard1, I am still not sure whether what I did here is better. Give it a look over.

Perhaps it's worse. I will have a fresh look tomorrow!

### CODE REVIEW

File: custom_components/adaptive_lighting/switch.py [hunk 1]
```diff
@@ -1165,43 +1164,45 @@ async def _update_manual_control_and_maybe_adapt(
             context.id,
         )
 
-        if adapt_brightness is None:
-            adapt_brightness = self.adapt_brightness_switch.is_on
-        if adapt_color is None:
-            adapt_color = self.adapt_color_switch.is_on
+        adapt_brightness = adapt_brightness or self.adapt_brightness_switch.is_on
```
Comment (th3w1zard1) [inline custom_components/adaptive_lighting/switch.py:1167]: Yeah I tried this change and ended up with errors in my test. This is your old code pulled from `_adapt_lights`

**Comment by th3w1zard1** [comment]
The way I wrote it just seemed cleaner: the logic flowed and made more sense. :shrug.

Looking at this it seems like it'd be difficult to modify if needed later down the line, which was the whole point of the small rewrite

**Review by th3w1zard1 — CHANGES_REQUESTED** [review]

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -1165,43 +1164,45 @@ async def _update_manual_control_and_maybe_adapt(
...
-                    force,
                     adapt_brightness,
                     adapt_color,
-                ):
+                    context,
+                )
+            )
+
+            if self._take_over_control and (manually_controlled or significant_change):
+                if manually_controlled:
```
[the remark this answers is not in the document]
Comment (basnijholt) [reply custom_components/adaptive_lighting/switch.py:1195]: > What was so bad about the way I wrote this func?


Nothing bad about it! I just personally find it fun to rewrite stuff in order to make the code simpler 😄

**Comment by th3w1zard1** [comment]
I've come around to liking this :)

**Review by th3w1zard1 — APPROVED** [review]

**Pull request merged**